### PR TITLE
For #5042 - Count total number of blocked trackers.

### PR DESCRIPTION
**Pull request by mcarare**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/browser/binding/BlockedTrackersBinding.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * Binding responsible for updating the total number of tracked blockers since install.
+ */
+class BlockedTrackersBinding(
+    store: BrowserStore,
+    private val tabId: String,
+    private val updateCount: (Int) -> Unit
+) : AbstractBinding<BrowserState>(store) {
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/browser/binding/BlockedTrackersBinding.kt:23]: A `Middleware` may be a better choice and independent from the UI. With the `TrackerBlockedAction` it's also easier to count them. :)

Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/browser/binding/BlockedTrackersBinding.kt:23]: I preferred to handle this locally in order to not interfere with release trains for AC and Focus. But If you think using middleware would improve this, I can give it a go tomorrow.

Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/browser/binding/BlockedTrackersBinding.kt:23]: I meant I can use the binding locally, I thought you suggested implementing a middleware in AC for this.

Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/browser/binding/BlockedTrackersBinding.kt:23]: In theory you could have a middleware on `BrowserStore` (implemented in `focus-android` for now), that counts every `TrackerBlockedAction`, persists the count, and also dispatches an action on `AppStore` (which you can use for reading the state in the UI).

**Comment by mcarare** [comment]
Closed in favor of https://github.com/mozilla-mobile/focus-android/pull/5049.

**Pull request closed without merging**